app/src/main/java/com/example/tasky/common/presentation/AgendaCard.kt

```diff
@@ -103,10 +118,25 @@ fun AgendaCard(
 }
 
 @Composable
-fun CustomCheckbox(isChecked: Boolean, color: Color, size: Dp) {
-    IconButton(onClick = { /*TODO*/ }) {
+fun CustomCheckbox(
+    agendaItem: AgendaItem? = null,
+    isTaskChecked: Boolean,
+    color: Color,
+    size: Dp,
+    toggleTaskComplete: () -> Unit = {},
+    setTaskCompleteForAgendaItemId: (String) -> Unit = { _ -> },
+    taskCompleteForAgendaItemId: String = ""
+) {
+    IconButton(onClick = {
+        if (agendaItem != null) {
+            if (agendaItem.cardType == AgendaItemType.Task) {
+                setTaskCompleteForAgendaItemId(agendaItem.id)
+                toggleTaskComplete()
+            }
+        }
+    }) {
```

Comment:
Instead of checking for the type in onClick, I'd probably rather set this button disabled for events and reminders, so you don't get the ripple there. Otherwise it might raise the impression that you could click on it, but it doesn't work